### DISCUSSION TITLE
v19.5.0-rc5

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3319] [Releases]
## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Highlights
This is a big release, that's why we decided to move away from the 19.4.x versioning and go to 19.5.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 19.5.0. Tests are more than welcome now but be extra careful with production environment.

- What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.
- PHP 7.4 is now the minimum required version and 8.2 is now supported
- M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- Support for Google Analytics 4 was added

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

## Changelog
There are no changes between 19.5.0-rc4 and 19.5.0-rc5.

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v19.5.0-rc5'>v19.5.0-rc5</a>.</em>

**Comment by djixas** [comment]
Any comment on this?

https://github.com/OpenMage/magento-lts/discussions/3265#discussioncomment-6034866

**Comment by lyll-de** [comment]
Hi, all

I'm new here and first want to say - thank you, guys, for you amazing work. 

I also have a question regarding 19.5.0-rc5 and newest 19.5.0. After I manually updated from stable OpenMage 19.4.23, I'm getting error on frontend: 

"Fatal error: Uncaught Error: Call to undefined function str_contains() in /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Helper/Security.php:40 Stack trace: #0 /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Model/Layout.php(344): Mage_Core_Helper_Security->validateAgainstBlockMethodBlacklist(Object(Mage_Page_Block_Html_Head), 'addJs', Array) #1 /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Model/Layout.php(210): Mage_Core_Model_Layout->_generateAction(Object(Mage_Core_Model_Layout_Element), Object(Mage_Core_Model_Layout_Element)) #2 /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Model/Layout.php(202): Mage_Core_Model_Layout->generateBlocks(Object(Mage_Core_Model_Layout_Element)) #3 /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Model/Layout.php(202): Mage_Core_Model_Layout->generateBlocks(Object(Mage_Core_Model_Layout_Element)) #4 /home/customer/www in /home/customer/www/mysite.com/web/releases/20230826123744/app/code/core/Mage/Core/Helper/Security.php on line 40"

I'm using PHP 7.4 and can't switch to 8.0 or higher due to another legacy extentions. assume it's because function str_contains() was introdused in PHP 8.0. 

Just don't know, that to do. Any advice would be appreciated

**Reply by Flyingmana** [reply]
Hi,

for many of the new methods there are polyfills available. Some as simple methods you can just copy into the project, and some as composer packages you can add as dependencies.

The first result of searching for str_contains brought me to https://gist.github.com/gdarko/5dbcf96dccb2a3eece6c10d084d008af which seems good enough.

I hope this info helps you a bit with finding a workaround

**Reply by lyll-de** [reply]
hi, 

thanks a lot

**Reply by fballiano** [reply]
@lyll-de we've added polyfills for everything so it's very weird that you've seen this error, did you try the 19.5.0 final? did you run "composer install"?

**Reply by lyll-de** [reply]
yes, I tried 19.5.0 final as well

no, I didn't run "composer install",  just copied files from zip archive over OpenMage 19.4.23

**Reply by fballiano** [reply]
in the zip you should have the vendor folder with some polyfills (and zend framework and other stuff)

**Reply by lyll-de** [reply]
thank you very much! it seems, that somehow I didn't pushed the vendor folder to test server. my mistake.
